youtube_dl/extractor/porntrex.py

```diff
@@ -0,0 +1,140 @@
+        'info_dict': {
+            'id': '311136',
+            'ext': 'mp4',
+            'title': 'Naomi gets fingered before the fucking',
+            'uploader': 'cumberland',
+            'description': 'Sexy brunette babe likes to get her tight cunt slammed in hardcore fashion.',
+            'age_limit': 18,
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        self.report_extraction(video_id)
+        private_string = 'Only active members can watch private videos.'
+        is_video_private_regex = re.compile(private_string)
+        if re.findall(is_video_private_regex, webpage):
+            self.raise_login_required()
+
+        title = self._html_search_regex(
+            r'<title>(.+?)</title>', webpage, 'title',)
+        movie_urls_regex = re.compile("'(https://www.porntrex.com/get_file/.*?)/'")
+        movie_urls = re.findall(movie_urls_regex, webpage)
+        uploader = self._search_regex(r'/members/\d+?/["\']>\s+(.+?)\s+</a>', webpage, 'new_uploader').strip()
+        thumbnails_regex = re.compile(r'href="(http.*?/screenshots/\d+.jpg/)"')
+        thumbnails_list = re.findall(thumbnails_regex, webpage)
+        thumbnails = []
+        for thumbs in thumbnails_list:
+            thumbnails.append({'url': thumbs})
+        formats = []
+        for movie_url in movie_urls:
+            formats.append({'url': movie_url,
+                            'ext': movie_url.split('.')[-1],
+                            'protocol': movie_url.split(':')[0],
+                            'height': int(self._search_regex(r'_(\d+)p.', movie_url.split('/')[-1], 'height', default='480')),
+                            })
+        self._sort_formats(formats)
+
+        return {
+            'id': video_id,
+            'title': title,
+            'description': self._og_search_description(webpage),
+            'uploader': uploader,
+            'thumbnails': thumbnails,
+            'formats': formats,
+            'age_limit': 18,
+        }
+
+
+class PornTrexPlayListIE(PornTrexBaseIE):
+    _NETRC_MACHINE = 'porntrex'
+    _VALID_URL = \
+        r'https?://(?:www\.)?porntrex\.com/playlists/(?P<id>[0-9]+)/'
+    _TEST = {
+        'url': 'https://www.porntrex.com/playlists/31075/2016-collection/',
+        'id': '31075',
+        'title': 'FTVGirls 2016 Collection',
+        'info_dict': {
+            'id': '345462',
+            'ext': 'mp4',
+            'uploader': 'publicgirls',
+            'title': 'FTVGirls.16.05 - Adria Part 2',
+            'description': 'https://www.indexxx.com/models/121033/adria-rae/',
+            'age_limit': 18,
+        }
+    }
+
+    def _real_extract(self, url):
+        playlist_id = self._match_id(url)
+        webpage = self._download_webpage(url, playlist_id)
+
+        get_all_urls_regex = re.compile('data-playlist-item="(.*?)"')
```

Comment:
Capturing empty string does not make any sense.

Reply:
This does not capture empty string. It gets all the playlist urls. They exist in a **data-playlist-item** attribute of an anchor element.

Reply:
The playlist urls appear on the playlist pages as **data-playlist-item** attributes of anchor elements.

Reply:
Again: it will capture empty string for `data-playlist-item=""`.

Reply:
Noted. Will improve the regex to reduce the chance for this using
`r'data-playlist-item=["\'](https?://www\.porntrex\.com/[^\'"]+)'`